subprojects/core/src/main/java/org/gradle/execution/plan/LocalTaskNodeExecutor.java

```diff
@@ -180,11 +190,18 @@ private static boolean missesDependency(Node producer, Node consumer) {
         return true;
     }
 
-    private void collectValidationProblem(Node producer, Node consumer, TypeValidationContext validationContext) {
+    private void collectValidationProblem(Node producer, Node consumer, TypeValidationContext validationContext, String consumerProducerPath) {
         TypeValidationContext.Severity severity = TypeValidationContext.Severity.WARNING;
         validationContext.visitPropertyProblem(
             severity,
-            String.format("Task '%s' uses the output of task '%s', without declaring an explicit dependency (using Task.dependsOn() or Task.mustRunAfter()) or an implicit dependency (declaring task '%s' as an input). This can lead to incorrect results being produced, depending on what order the tasks are executed", consumer, producer, producer)
+            String.format("Gradle detected a problem with the following location: '%s'. " +
+                    "Task '%s' uses this output of task '%s', without declaring an explicit dependency (using Task.dependsOn() or Task.mustRunAfter()) or an implicit dependency (declaring task '%s' as an input). "
+                + "This can lead to incorrect results being produced, depending on what order the tasks are executed",
```

Comment:
The `+` characters should be consistently placed here. I also think that there is no need for a comma before `without` in the message.

Reply:
Done.